Assets/Scripts/Crab.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Crab : MonoBehaviour
+{
+    const string LEFT = "left";
+    const string RIGHT = "right";
+
+    [SerializeField]
+    Transform _castPos;
+    [SerializeField]
+    float _baseHorizontalCastDist;
+    [SerializeField]
+    float _baseVerticalCastDist;
+    [SerializeField]
+    float playerFollowDist;
+    [SerializeField]
+    Transform _playersTransform;
```

Comment:
_playerTransform shouldn't be assigned by hand, making sure that each enemy finds player on its own is better approach.
Add to the start: `_playerTransform = GameObject.FindGameObjectWithTag("Player").transform` using Find is very inefficient, but it is okay to use once per enemy.